File: src/main/java/com/wine/to/up/am/parser/service/logging/AmServiceNotableEvents.java
package com.wine.to.up.am.parser.service.logging;

import com.wine.to.up.commonlib.logging.NotableEvent;

public enum AmServiceNotableEvents implements NotableEvent {

    SOME_DEMO_EVENT("Something happened");

    private final String template;

    AmServiceNotableEvents(String template) {
        this.template = template;
    }

    public String getTemplate() {
        return template;
    }

    @Override
    public String getName() {
        return name();
    }


}

File: src/main/java/com/wine/to/up/am/parser/service/logging/AmServiceNotableEvents.kt
package com.wine.to.up.am.parser.service.logging

import com.wine.to.up.commonlib.logging.NotableEvent

enum class AmServiceNotableEvents(private val template: String) : NotableEvent {
    SOME_DEMO_EVENT("Something happened");

    override fun getTemplate(): String {
        return template
    }

    override fun getName(): String {
        return name
    }
}